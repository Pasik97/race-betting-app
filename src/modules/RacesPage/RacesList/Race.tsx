import React from 'react';
import * as P from './parts';
import * as TP from 'components/BasicTable/parts';
import { RaceWithBet } from 'store/races/constants';

interface RaceProps {
   race: RaceWithBet;
}

const Race: React.FC<RaceProps> = ({ race }) => (
   <TP.Row key={race.id}>
      <TP.Cell><P.StyledLink to={'/races/' + race.id}>{race.id}</P.StyledLink></TP.Cell>
      <TP.Cell><P.StyledLink to={'/races/' + race.id}>{race.name}</P.StyledLink></TP.Cell>
      <TP.Cell><TP.Label>{race.active ? 'Active' : 'Inactive'}</TP.Label></TP.Cell>
   </TP.Row>
);

export default Race;
