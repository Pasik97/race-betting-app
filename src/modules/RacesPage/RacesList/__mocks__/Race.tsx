const mock = () => '(Race /)';

export default mock;
