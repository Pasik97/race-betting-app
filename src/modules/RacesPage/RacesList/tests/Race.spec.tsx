import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Race from "../Race";
import { mockRace, secondMockRace } from "store/races/tests/mockData";

describe('<Race />', () => {
   it('should render race id and name as link to single race page and "Active" label if race is active', () => {
      const component = create(
         <BrowserRouter>
            <Race race={mockRace} />
         </BrowserRouter>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should render race id and name as link to single race page and "Inactive" label if race is inactive', () => {
      const component = create(
         <BrowserRouter>
            <Race race={secondMockRace} />
         </BrowserRouter>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });
});
