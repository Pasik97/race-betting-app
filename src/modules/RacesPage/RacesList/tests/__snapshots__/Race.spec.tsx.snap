// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Race /> should render race id and name as link to single race page and "Active" label if race is active 1`] = `
<div
  className="sc-gKclnd huABHg"
>
  <div
    className="sc-eCImPb jHOpYB"
  >
    <a
      className="sc-gsDKAQ kkNNhb"
      href="/races/1"
      onClick={[Function]}
    >
      1
    </a>
  </div>
  <div
    className="sc-eCImPb jHOpYB"
  >
    <a
      className="sc-gsDKAQ kkNNhb"
      href="/races/1"
      onClick={[Function]}
    >
      mock race
    </a>
  </div>
  <div
    className="sc-eCImPb jHOpYB"
  >
    <p
      className="sc-iCfMLu iDUcVZ"
    >
      Active
    </p>
  </div>
</div>
`;

exports[`<Race /> should render race id and name as link to single race page and "Inactive" label if race is inactive 1`] = `
<div
  className="sc-gKclnd huABHg"
>
  <div
    className="sc-eCImPb jHOpYB"
  >
    <a
      className="sc-gsDKAQ kkNNhb"
      href="/races/2"
      onClick={[Function]}
    >
      2
    </a>
  </div>
  <div
    className="sc-eCImPb jHOpYB"
  >
    <a
      className="sc-gsDKAQ kkNNhb"
      href="/races/2"
      onClick={[Function]}
    >
      second mock race
    </a>
  </div>
  <div
    className="sc-eCImPb jHOpYB"
  >
    <p
      className="sc-iCfMLu iDUcVZ"
    >
      Inactive
    </p>
  </div>
</div>
`;
